src/app/pages/camera.cpp

```diff
@@ -27,25 +27,91 @@ CameraPage::CameraPage(QWidget *parent) : QWidget(parent)
 
     connect(this, &CameraPage::disconnected, [layout,this]() {
         layout->setCurrentIndex(0);
-        if (this->config->get_cam_autoconnect()) {
-            qDebug() << "Camera disconnected. Auto reconnect in" << this->config->get_cam_autoconnect_time_secs() << "seconds";
-            this->reconnect_message = this->status->text() + ". reconnecting in %1 ";
-            this->reconnect_in_secs = this->config->get_cam_autoconnect_time_secs();
-            this->reconnect_timer->start(1000);
-        }
+        this->disconnect_stream();
     });
-    connect(this, &CameraPage::connected_local, [layout]() { layout->setCurrentIndex(1); });
-    connect(this, &CameraPage::connected_network, [layout]() { layout->setCurrentIndex(2); });
+    connect(this, &CameraPage::connected_local, [layout, this]() { layout->setCurrentIndex(1); this->connected=true; });
+    connect(this, &CameraPage::connected_network, [layout, this]() { layout->setCurrentIndex(2); this->connected=true; });
+
+
+    videoContainer_ = nullptr;
+    videoWidget_ = nullptr;
+
+    // Write camera_overlay to /tmp so that we can later point gstreamer to it
```

Comment:
assuming gstreamer need the file to exist somewhere on the fs?

Reply:
Right, the overlay pipeline requires it to be on disk